src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,185 @@
+--- 
+title: "State restoration for Android apps"
+description: "How to restore the state of your Android app after it's been killed by the OS."
+---
+
+When a user runs a mobile app and then selects another
+app to run, the first app is moved to the background,
+or _backgrounded_. The operating system (both iOS and Android)
+typically kill the backgrounded app to release memory and
+improve performance for the app running in the foreground.
+
+When the user selects the app again, bringing it
+back to the foreground, the OS relaunches it.
+But, unless you've set up a way to save the
+state of the app before it was killed,
+you've lost the state and the app starts from scratch.
+The user has lost the continuity they expect,
+which is clearly not ideal.
+(Imagine filling out a lengthy form and being interrupted
+by a phone call _before_ clicking **Submit**.)
+
+So, how can you restore the state of the app so that
+it looks like it did before it was sent to the
+background?
+
+Flutter has a solution for this with the
+[`RestorationManager`][] (and related classes)
+in the [services][] library.
+With the `RestorationManager`, the Flutter framework
+provides the state data to the engine _as the state
+changes_, so that the app is ready when the OS signals
+that it's about to kill the app, giving the app only
+moments to prepare.
+
+{{site.alert.secondary}}
+  **Instance state vs long-lived state**
+  When should you use the `RestorationManager` and
+  when should you save state to long term storage?
+  _Instance state_
+  (also called _short-term_ or _ephemeral_ state),
+  includes unsubmitted form field values, the currently
+  selected tab, and so on. On Android, this is
+  limited to 1 MB and, if the app exceeds this,
+  it crashes with a `TransactionTooLargeException`
+  error in the native code.
+{{site.alert.end}}
+
+[state]: {{site.url}}/development/data-and-backend/state-mgmt/ephemeral-vs-app
+
+## Overview
+
+You can enable state restoration with just a couple tasks:
+
+1. Define a `restorationId` or a `restorationScopeId`
+   for all widgets that support it,
+   such as [`TextField`][] and [`ScrollView`][].
+   This automatically enables built-in state restoration
+   for those widgets.
+
+2. For custom widgets,
+   you must decide what state you want to restore
+   and hold that state in a [`RestorableProperty`][].
+   (The Flutter API provides various subclasses for
+   different data types.)
+   Define those `RestorableProperty` widgets 
+   in a `State` class that uses the [`RestorationMixin`][].
+   Register those widgets with the mixin in a
+   `restoreState` method.
+
```

Comment:
Add as third step: If you use any Navigator API (like push, pushNamed, etc) migrate to the API that has restorable in the name (e.g. restorablePush, resstorablePushNamed, etc) to restore the navigation stack.

Reply:
You already cover this in the "Restoring navigation state" section, but may be good to also have that included in the overview here...